src/v/archival/ntp_archiver_service.cc

```diff
@@ -1359,6 +1374,7 @@ ss::future<ntp_archiver::upload_group_result> ntp_archiver::wait_uploads(
         } else {
             // We have flushed projected clean offset if it was set
             _projected_manifest_clean_at.reset();
+            _last_marked_clean_time = ss::lowres_clock::now();
```

Comment:
Good point.  This wouldn't break anything because if the main upload loop is active we'll be flushing anyway, but it's clearer to make it conditional, so I've done that.